ecs-logs-collector.sh

```diff
@@ -460,13 +460,13 @@ enable_docker_debug()
   case "${os_name}" in
     amazon)
 
-      if [ -e /etc/sysconfig/docker ] && grep -q "OPTIONS=\"-D" /etc/sysconfig/docker
+      if [ -e /etc/sysconfig/docker ] && grep -q "^OPTIONS=\"-D \$OPTIONS" /etc/sysconfig/docker
```

Comment:
I don't think we want the `\$OPTIONS` bit here. If the user hasn't done a similar appending as line 469, it will fail. Separately, I don't think we want to restrict ourselves to the line having to start with `OPTIONS`. Maybe `grep -q "^\s*OPTIONS=...` So we can match even if there is whitespace at the beginning of the line.

Reply:
My intention was to check whether this script had enabled debug previously, or not. Even without the $OPTIONS bit, this grep check will only find instances where -D is at the beginning of the OPTIONS value. "^\s*OPTIONS=\".*\s*-D" would catch the debug switch even if placed after other options but risks a false positive if -D appears in a string for another option. I have done some testing and found that multiple instances of -D in the OPTIONS do not seem to have any adverse effects. so sticking with "^OPTIONS=\"-D" seems to make the most sense.

Good point. Checking for white space is worth doing.